### PR TITLE
#160618106 comment on articles and reply on comment

### PR DESCRIPTION
**Pull request by tonyguesswho**
### What does this PR do?
This PR shows how articles controller is created which performs the CRUD functions
### Description of Task to be completed?
Create comment controller which performs CRUD functions
- enables a user to comment on an article
- enables a user to edit comment on an article
- enables a user to update comment on an article
- enables a user to delete the comment on an article
- enables a user to get all comments and corresponding replies
- enables a user to reply on a comment
- enables a user to edit reply on a comment
- enables a user to update reply on a comment
- enables a user to delete reply on a comment
- enables a user to get all replies associated with a comment
How should this be manually tested?
- Clone the project
- Run `npm install` to install dependencies
- Run `sequelize db:migrate` to migrate the database
- `npm run test`

Any background context you want to provide?
None
What are the relevant pivotal tracker stories?
#160618106

Screenshots (if appropriate)
None
Questions:
None

### CODE REVIEW

File: controllers/commentController.js [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+const noComment = {
+  error: true,
+  message: 'Comment does not exist'
+};
+
+const noReply = {
+  error: true,
+  message: 'Reply does not exist'
+};
+
```
Comment (AdinoyiSadiq) [inline controllers/commentController.js]: Hi. Could we move these error messages to a helper file and import them instead? Maybe we could call it something like 'errorMessages'. What do you think?

**Review by AdinoyiSadiq — APPROVED** [review]
This looks good. Good job.

File: controllers/commentController.js (hunk 1 above)
Comment (tonyguesswho) [reply controllers/commentController.js]: That's fine.

**Comment by Phunmbi** [comment]
Nice Job abstracting the error messages. LGTM

File: lib/helper.js [hunk 2]
```diff
@@ -3,37 +3,48 @@ const db = require('../models');
...
-const findUser = async (email) => {
-  const existingUser = await User.findOne({ where: { email } });
-  return existingUser;
+/**
+ * checks for the existence of any data in the database
+ * @param {object} model The database model.
+ * @param {object} searchParam The search parameter needed to query the database.
+ * @returns {boolean} existing
+ */
+const findItem = async (model, searchParam) => {
```
Comment (Phunmbi) [inline lib/helper.js]: Nice idea to abstract this and make it reusable

**Review by Phunmbi — APPROVED** [review]

**Pull request merged**